services/ux-backend/main.go

```diff
@@ -18,37 +18,34 @@ type serverConfig struct {
 	tlsEnabled           bool
 }
 
+func readEnvVar[T any](envVarName string, defaultValue T, parser func(str string) (T, error)) (T, error) {
+	if str := os.Getenv(envVarName); str == "" {
+		klog.Infof("no user-defined %s provided, defaulting to %v", envVarName, defaultValue)
+		return defaultValue, nil
+	} else if value, err := parser(str); err != nil {
+		return *new(T), fmt.Errorf("malformed user-defined %s value %v: %v", envVarName, value, err)
```

Comment:
In case of an error the value always will return 0, so let's not print the value in Errorf.

Reply:
1. It will not return 0 it will return the default value of the type: `0` for int, `false` for bool, `nil` for pointers, etc.
2. There is a bug here. I should point `str` and `value` because I want to show the faulty string. I will fix it.